server/webapp/WEB-INF/rails/webpack/views/components/table/index.tsx

```diff
@@ -88,11 +98,59 @@ class TableHeader extends MithrilViewComponent<HeaderAttrs> {
   }
 }
 
-export class Table extends MithrilViewComponent<Attrs> {
-  view(vnode: m.Vnode<Attrs>) {
-    return <table className={styles.table} data-test-id={vnode.attrs["data-test-id"] || "table"}>
+export class Table extends MithrilComponent<Attrs, State> {
+
+  oninit(vnode: m.Vnode<Attrs, State>): any {
+
+    // Return if not draggable
+    if (!vnode.attrs.draggable) {
+      return;
+    }
+
+    vnode.state.dragStart = (e) => {
+      vnode.state.dragged          = Number(e.currentTarget.dataset.id);
+      vnode.state.dragging         = vnode.state.dragged;
+      e.dataTransfer.effectAllowed = "move";
+      e.dataTransfer.setData("text/html", null);
+    };
+
+    vnode.state.dragOver = (e) => {
+      e.preventDefault();
+
+      const toBeReplaced                 = e.target;
+      const updatedPositionWhileDragging = vnode.state.dragging;
+      const newPosition                  = Number(toBeReplaced.dataset.id) || updatedPositionWhileDragging;
+
+      if (updatedPositionWhileDragging === newPosition) {
+        return;
+      }
+
+      vnode.attrs.data.splice(newPosition, 0, vnode.attrs.data.splice(updatedPositionWhileDragging, 1)[0]);
```

Comment:
Not easy to understand. Please can you use the appropriate variable name and pull out `vnode.attrs.data.splice(updatedPositionWhileDragging, 1)[0]` in separate variable